### PR TITLE
Api provide restriction for landlord users to create only two listing and subscribers to create unlimited listings

### PR DESCRIPTION
**Pull request by Odeane**
### User Story
```
As an api
in order to prevent more than two listing
i would liked to restrict users creation ability to the above.
```
## Changes proposed in this pull request:
* Registered users can only create two articles.
* Subscribers can create unlimited user.

### CODE REVIEW

File: spec/requests/api/v1/listings/listing_limitation_spec.rb [hunk 1]
```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+RSpec.describe 'POST /api/v1/listings', type: :request do
+  let!(:landlord) { create(:user, role: 'registered') }
+  let!(:listing) { 5.times { create(:listing, landlord: landlord) } }
```
Comment (viktorsommar) [inline spec/requests/api/v1/listings/listing_limitation_spec.rb]: Why create 5 listings for the registered user?

**Review by viktorsommar — CHANGES_REQUESTED** [review]

**Review by Alex-Bellamy — CHANGES_REQUESTED** [review]
not sure why 5 listings are generated

File: spec/requests/api/v1/listings/listing_limitation_spec.rb (hunk 1 above)
Comment (Odeane) [reply spec/requests/api/v1/listings/listing_limitation_spec.rb]: Good point

File: app/controllers/api/v1/listings_controller.rb [hunk 2]
```diff
@@ -22,24 +22,24 @@ def create
...
-    
     params_images = params[:listing][:images]
     if params_images.present?
-      params_images.each do |image| 
-        DecodeService.attach_image(image, listing.image) 
+      params_images.each do |image|
+        DecodeService.attach_image(image, listing.images)
       end
-      
+
```
Comment (viktorsommar) [inline app/controllers/api/v1/listings_controller.rb]: extra line

File: app/controllers/api/v1/listings_controller.rb [hunk 3]
```diff
@@ -22,24 +22,24 @@ def create
...
     elsif !attach_image(listing)
       listing.destroy
-      render json: { message: "The image can't be blank"}, status:422
-    else 
+      render json: { message: "The image can't be blank" }, status: 422
+    else
       render_error_message(listing.errors)
     end
   rescue StandardError => e
+
```
Comment (viktorsommar) [inline app/controllers/api/v1/listings_controller.rb]: formatting

File: app/controllers/api/v1/subscriptions_controller.rb [hunk 4]
```diff
@@ -1,30 +1,30 @@
...
-      end
-    rescue => error
-      render_stripe_error(error.message)
-    end 
+    customer_id = get_customer(params[:stripeToken])
+
+    subscription = Stripe::Subscription.create({
+                                                 customer: customer_id,
+                                                 plan: 'bds_subscription_plan'
+                                               })
```
Comment (viktorsommar) [inline app/controllers/api/v1/subscriptions_controller.rb:14]: what happened here with formatting?

**Review by Alex-Bellamy — APPROVED** [review]

File: spec/requests/api/v1/listings/listing_limitation_spec.rb [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+             }
+           }, headers: landlord_headers
+    end
+
+    it 'is expected to return 422 response status' do
+      expect(response).to have_http_status 422
+    end
+
+    it 'is expect to return error message when creating third listing' do
+      expect(response_json['message']).to eq 'Subscribe to create more listing.'
```
Comment (viktorsommar) [inline spec/requests/api/v1/listings/listing_limitation_spec.rb]: very minor but it should be "... more listingS."

File: spec/requests/api/v1/listings/listing_limitation_spec.rb [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+    end
+
+    it 'is expected to have six listings created' do
+      expect(Listing.all.count).to eq 2
```
Comment (viktorsommar) [inline spec/requests/api/v1/listings/listing_limitation_spec.rb:52]: the tests says expected 6 listings but the actual expectations is equal to 2?

File: spec/requests/api/v1/listings/listing_limitation_spec.rb [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+    it 'is expected to return success message' do
+      expect(response_json['message']).to eq 'The listing has been created successfully!'
+    end
+
+    it 'is expected to have two listings created' do
+      expect(Listing.all.count).to eq 2
+    end
+  end
+
+  describe 'subscriber can successfully create more then two listings' do
```
Comment (viktorsommar) [inline spec/requests/api/v1/listings/listing_limitation_spec.rb]: than

File: spec/requests/api/v1/listings/listing_limitation_spec.rb [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+    let!(:landlord_credentials) { landlord.create_new_auth_token }
+    let!(:landlord_headers) { { HTTP_ACCEPT: 'application/json' }.merge!(landlord_credentials) }
+
+    let!(:listing) { create(:listing, landlord: landlord) }
+
+    before do
+      post '/api/v1/listings',
+           params: {
+             listing: {
+               category: 'From Params',
```
Comment (viktorsommar) [inline spec/requests/api/v1/listings/listing_limitation_spec.rb]: what category is this?

**Review by viktorsommar — CHANGES_REQUESTED** [review]

File: app/controllers/api/v1/subscriptions_controller.rb (hunk 4 above)
Comment (Odeane) [reply app/controllers/api/v1/subscriptions_controller.rb:14]: Thats the formatting the ruby formatter gave

File: spec/requests/api/v1/listings/listing_limitation_spec.rb (hunk 5 above)
Comment (Odeane) [reply spec/requests/api/v1/listings/listing_limitation_spec.rb]: I admire your control of the english language Viktor.
I appreciate your linguistics skills.

**Review by viktorsommar — APPROVED** [review]

**Review by Alex-Bellamy — APPROVED** [review]

**Pull request merged**